weed/mount/weedfs_dir_read.go

```diff
@@ -182,37 +177,60 @@ func (wfs *WFS) doReadDirectory(input *fuse.ReadIn, out *fuse.DirEntryList, isPl
 			}
 			wfs.outputFilerEntry(entryOut, inode, entry)
 		}
-		dh.lastEntryName = entry.Name()
 		return true
 	}
 
-	entryChan := make(chan *filer.Entry, 128)
-	var err error
-	go func() {
-		if err = meta_cache.EnsureVisited(wfs.metaCache, wfs, dirPath, entryChan); err != nil {
-			glog.Errorf("dir ReadDirAll %s: %v", dirPath, err)
+	if input.Offset < directoryStreamBaseOffset {
+		if !isPlusMode {
+			if input.Offset == 0 {
+				out.AddDirEntry(fuse.DirEntry{Mode: fuse.S_IFDIR, Name: "."})
+			}
+			out.AddDirEntry(fuse.DirEntry{Mode: fuse.S_IFDIR, Name: ".."})
+		} else {
+			if input.Offset == 0 {
+				out.AddDirLookupEntry(fuse.DirEntry{Mode: fuse.S_IFDIR, Name: "."})
+			}
+			out.AddDirLookupEntry(fuse.DirEntry{Mode: fuse.S_IFDIR, Name: ".."})
 		}
-		close(entryChan)
-	}()
-	hasData := false
-	for entry := range entryChan {
-		hasData = true
-		processEachEntryFn(entry, false)
-	}
-	if err != nil {
-		return fuse.EIO
+		input.Offset = directoryStreamBaseOffset
 	}
 
-	if !hasData {
-		listErr := wfs.metaCache.ListDirectoryEntries(context.Background(), dirPath, dh.lastEntryName, false, int64(math.MaxInt32), func(entry *filer.Entry) bool {
-			return processEachEntryFn(entry, false)
-		})
-		if listErr != nil {
-			glog.Errorf("list meta cache: %v", listErr)
-			return fuse.EIO
+	var lastEntryName string
+	if input.Offset >= directoryStreamBaseOffset {
```

Comment:
this If condition seems to be always true

Reply:
this commit fixes that: https://github.com/chrislusf/seaweedfs/pull/3102/commits/1fabbe8a25f8cd4183589f728dcf9a5031905b2d